### PR TITLE
fix(discover2): Be able to apply aggregate functions on tags with hyphens

### PR DESCRIPTION
**Pull request by dashed**
I added a test with events with tags that makes use of special characters: period, underscore, and hyphen.

### CODE REVIEW

**Review by evanh — APPROVED** [review]

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -847,7 +847,7 @@ def to_list(value):
     "sum": {"snuba_name": "sum", "fields": ["transaction.duration"]},
 }
 
-AGGREGATE_PATTERN = re.compile(r"^(?P<function>[^\(]+)\((?P<column>[a-z\._]*)\)$")
+AGGREGATE_PATTERN = re.compile(r"^(?P<function>[^\(]+)\((?P<column>[a-z\._-]*)\)$")
```
Comment (dashed) [inline src/sentry/api/event_search.py]: @dashed columns can have numbers

Comment (lynnagara) [reply src/sentry/api/event_search.py]: Tag keys can have any character

Comment (markstory) [reply src/sentry/api/event_search.py]: We should update the patterns used here to accept the full character range. Does the pattern in the client side code need to be updated as well?

Comment (dashed) [reply src/sentry/api/event_search.py]: I'll check the client side.

**Comment by dashed** [comment]
Snuba borks on `SchemaValidationError("['tags[sub_customer.is-enterprise)]'] is not valid under any of the given schemas",)` which will need to be addressed.

File: src/sentry/api/event_search.py (hunk 1 above)
Comment (dashed) [reply src/sentry/api/event_search.py]: Snuba has a schema validation step on the tag names: https://github.com/getsentry/snuba/blob/5e5de9e85b68fb931c7e9711db5c873774363dcf/snuba/query/schema.py#L117-L127

I'm going to update based on this.

File: src/sentry/static/sentry/app/views/eventsV2/utils.tsx [hunk 2]
```diff
@@ -222,9 +222,8 @@ export function getAggregateAlias(field: string): string {
   }
   return field
     .replace(AGGREGATE_PATTERN, '$1_$2')
-    .replace('.', '_')
-    .replace(/_+$/, '')
-    .toLowerCase();
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/eventsV2/utils.tsx]: @markstory i removed this

File: src/sentry/static/sentry/app/views/eventsV2/utils.tsx [hunk 3]
```diff
@@ -222,9 +222,8 @@ export function getAggregateAlias(field: string): string {
...
+    .replace(/\./g, '_')
+    .replace(/_+$/, '');
```
Comment (dashed) [inline src/sentry/static/sentry/app/views/eventsV2/utils.tsx]: @markstory according to docs,  if `pattern` is a string, only the first occurrence will be replaced. I made this a regex.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

Comment (dashed) [reply src/sentry/static/sentry/app/views/eventsV2/utils.tsx]: Oops. This was for line 225

**Review by markstory — APPROVED** [review]

**Pull request merged**